source/PlayerInfo.cpp

```diff
@@ -587,9 +603,8 @@ void PlayerInfo::AddChanges(list<DataNode> &changes)
 		}
 	}
 
-	// Only move the changes into my list if they are not already there.
-	if(&changes != &dataChanges)
-		dataChanges.splice(dataChanges.end(), changes);
+	// The dataChanges list is no longer added to. All events whose changes are
+	// applied are added to the pastEvents list.
```

Comment:
I just realized that we actually do need to add to dataChanges if the event was stored in the save file without a name from prior to this change. Otherwise the event's changes will be lost when the save is reloaded.